tools/swift-reflection-dump/swift-reflection-dump.cpp

```diff
@@ -32,200 +33,227 @@
 #endif
 
 #include <algorithm>
-#include <iostream>
 #include <csignal>
+#include <iostream>
 
+using llvm::ArrayRef;
 using llvm::dyn_cast;
 using llvm::StringRef;
-using llvm::ArrayRef;
 using namespace llvm::object;
 
 using namespace swift;
 using namespace swift::reflection;
 using namespace swift::remote;
 using namespace Demangle;
 
-enum class ActionType {
-  DumpReflectionSections,
-  DumpTypeLowering
-};
+enum class ActionType { DumpReflectionSections, DumpTypeLowering };
 
 namespace options {
-static llvm::cl::opt<ActionType>
-Action(llvm::cl::desc("Mode:"),
-       llvm::cl::values(
-         clEnumValN(ActionType::DumpReflectionSections,
-                    "dump-reflection-sections",
-                    "Dump the field reflection section"),
-         clEnumValN(ActionType::DumpTypeLowering,
-                    "dump-type-lowering",
-                    "Dump the field layout for typeref strings read from stdin")),
-       llvm::cl::init(ActionType::DumpReflectionSections));
+static llvm::cl::opt<ActionType> Action(
+    llvm::cl::desc("Mode:"),
+    llvm::cl::values(
+        clEnumValN(ActionType::DumpReflectionSections,
+                   "dump-reflection-sections",
+                   "Dump the field reflection section"),
+        clEnumValN(
+            ActionType::DumpTypeLowering, "dump-type-lowering",
+            "Dump the field layout for typeref strings read from stdin")),
+    llvm::cl::init(ActionType::DumpReflectionSections));
 
 static llvm::cl::list<std::string>
-BinaryFilename("binary-filename", llvm::cl::desc("Filenames of the binary files"),
-               llvm::cl::OneOrMore);
+    BinaryFilename("binary-filename",
+                   llvm::cl::desc("Filenames of the binary files"),
+                   llvm::cl::OneOrMore);
 
 static llvm::cl::opt<std::string>
-Architecture("arch", llvm::cl::desc("Architecture to inspect in the binary"),
-             llvm::cl::Required);
+    Architecture("arch",
+                 llvm::cl::desc("Architecture to inspect in the binary"),
+                 llvm::cl::Required);
 } // end namespace options
 
-template<typename T>
-static T unwrap(llvm::Expected<T> value) {
+template <typename T> static T unwrap(llvm::Expected<T> value) {
   if (value)
     return std::move(value.get());
-  std::cerr << "swift-reflection-test error: " << toString(value.takeError()) << "\n";
+  llvm::errs() << "swift-reflection-test error: " << toString(value.takeError())
+               << "\n";
+  exit(EXIT_FAILURE);
+}
+
+static void reportError(std::error_code EC) {
+  assert(EC);
+  llvm::errs() << "swift-reflection-test error: " << EC.message() << ".\n";
   exit(EXIT_FAILURE);
 }
 
-using NativeReflectionContext
-    = swift::reflection::ReflectionContext<External<RuntimeTarget<sizeof(uintptr_t)>>>;
+using NativeReflectionContext =
+    ReflectionContext<External<RuntimeTarget<sizeof(uintptr_t)>>>;
```

Comment:
I think that this is broken with MSVC (ambiguous reference to `ReflectionContext`).